src/ResourceManager/Compute/Commands.Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -290,5 +297,19 @@ public string Location
                 asyncCmdlet.WriteObject(psObject);
             }
         }
+
+        private VirtualMachineScaleSetIdentity GetVmssIdentityFromArgs()
+        {
+            var isUserAssignedEnabled = !string.IsNullOrWhiteSpace(UserAssignedIdentity);
+            return SystemAssignedIdentity || isUserAssignedEnabled
+                ? new VirtualMachineScaleSetIdentity
+                {
+                    Type =  !isUserAssignedEnabled ? ResourceIdentityType.SystemAssigned
+                        : SystemAssignedIdentity ? ResourceIdentityType.SystemAssignedUserAssigned
+                        : ResourceIdentityType.UserAssigned,
+                    IdentityIds = isUserAssignedEnabled ? new[] { UserAssignedIdentity } : null,
+                }
+                : null;
```

Comment:
@praries880 nit: I might refactor this return statement so that it includes an if-statement

```cs
if (SystemAssignedIdentity.IsPresent || isUserAssignedEnabled)
{
    return new VirtualMachineScaleSetIdentity
    {
        Type =  !isUserAssignedEnabled ? 
                        ResourceIdentityType.SystemAssigned :
                        SystemAssignedIdentity ?
                            ResourceIdentityType.SystemAssignedUserAssigned :
                            ResourceIdentityType.UserAssigned,
         IdentityIds = isUserAssignedEnabled ? new[] { UserAssignedIdentity } : null
    }
}

return null;
```

Reply:
@cormacpayne  is this for readability? The current code does make sure that if the top level conditions are not met we return a null value...

Reply:
added comment on the function